File: Spring Boot/microservices/distance-service/src/main/java/pl/edu/pg/benchmarking/distance/dto/DistanceResponse.java
package pl.edu.pg.benchmarking.distance.dto;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import pl.edu.pg.benchmarking.distance.Point;

import java.util.LinkedList;
import java.util.List;
import java.util.function.BiFunction;

@Getter
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class DistanceResponse {

    private int placeId1;

    private int placeId2;

    private double distance;

    public static BiFunction<List<Point>, List<Double>, List<DistanceResponse>> entityToDtoMapper() {
        return (points, distances) -> {
            List<DistanceResponse> distanceResponseList = new LinkedList<>();
            for (int iter = 0; iter < points.size(); iter++) {
                distanceResponseList.add(
                        new DistanceResponse(points.get(iter).getPlaceId1(), points.get(iter).getPlaceId2(), distances.get(iter)));
            }
            return distanceResponseList;
        };
    }
}

File: Dotnet/microservices/distance-service/Responses/DistanceResponse.cs
using System.Text.Json.Serialization;
using DistanceService.Models;

namespace DistanceService.Responses
{
    public class DistanceResponse
    {
        [JsonPropertyName("placeId1")]
        public int PlaceId1 { get; set; }

        [JsonPropertyName("placeId2")]
        public int PlaceId2 { get; set; }

        [JsonPropertyName("distance")]
        public double Distance { get; set; }

        private DistanceResponse(int placeId1, int placeId2, double distance)
        {
            PlaceId1 = placeId1;
            PlaceId2 = placeId2;
            Distance = distance;
        }

        public static List<DistanceResponse> EntityToDtoMapper(List<Point> points, List<double> distances)
        {
            List<DistanceResponse> distanceResponseList = new List<DistanceResponse>();
            for (int iter = 0; iter < points.Count; iter++)
            {
                distanceResponseList.Add(
                    new DistanceResponse(points[iter].PlaceId1, points[iter].PlaceId2, distances[iter]));
            }
            return distanceResponseList;
        }
    }
}
